proxy/src/telemetry/metrics/mod.rs

```diff
@@ -174,22 +180,33 @@ impl Root {
 
     fn request(&mut self, labels: RequestLabels) -> &mut http::RequestMetrics {
         self.requests.scopes.entry(labels)
-            .or_insert_with(http::RequestMetrics::default)
+            .or_insert_with(|| http::RequestMetrics::default().into())
+            .stamped()
```

Comment:
So, each of these will double stamp on the first insertion, right? Since the creation by `into`, and then the returned entry will have `stamped` called on it.

I also noticed that you added a `Default` impl for `Stamped<T>`, but then don't use it because a closure is used here instead.

Reply:
yes, that's right.

i can probably remove the default impl, or i can use `or_insert_with(Stamped::default)` though that's less clear imo.